facturx/facturx.py

```diff
@@ -147,3 +147,129 @@ def write_xml(self, path):
         with open(path, 'wb') as f:
             f.write(self.xml_str)
 
+    def __make_dict(self):
+        field_file = os.path.join(os.path.dirname(__file__), 'flavors\\fields.yml')
+        req_list = []
+        with open(field_file, 'r') as f:
+            data = f.read()
+            dict_data = yaml.load(data)
+
+        for k, v in dict_data.items():
+            for kv, vv in v.items():
+                if kv == '_required':
+                    if vv is True:
+                        req_list.append(k)
+
+        factx_value = []
+        factx_key = []
+        zugf_key = []
+        zugf_value = []
+
+        for k, v in dict_data.items():
```

Comment:
Could you achieve the same thing with fewer nested IFs?

Reply:
Can you suggest something? The problem is I am looping after every IF.